### ISSUE TITLE
About models validation based on MRR

### ISSUE THREAD

**Issue by loreloc** [facebookresearch/ssl-relation-prediction#16]
Hi, I am trying your code an I think there is something wrong about the validation step in the training loop.

That is, consider the line at https://github.com/facebookresearch/ssl-relation-prediction/blob/main/src/engines.py#L197.
At this line, the value assigned to the variable ```split``` is actually the last value that was assigned in the previous loop for computing step-wise metrics over all the splits (L.192-195).
The last value is in fact ```"test"```.

So I think that at the end the model is wrongly validated against the test set instead of the validation set.

**Comment by yihong-chen** [comment]
@loreloc Thanks for reporting this. It is very helpful to us. It is indeed my fault. I accidentally removed a line when I migrated the code from our internal repo to the public repo.  At that time we were also exploring some meta-learning algorithms.
```
split = 'meta_valid' if 'meta_valid' in self.dataset.splits else 'valid'
```
However, the results we reported in the paper and in the readme were evaluated on the test set.

**Comment by yihong-chen** [comment]
Hi @loreloc I just updated the repo. Let me if you still have any issues

**Comment by loreloc** [comment]
Hi @yihong-chen. I am glad this was helpful :) Good work !
